cypress/support/step_definitions/index.ts

```diff
@@ -129,84 +128,57 @@ Then('the URL contains {string}', (str: string) => {
+  const xhr = client.history.filter((item: HistoryEntry) => item.url.pathname === url)
+  // If we are asserting the URL _wasn't_ called, and we don't find it, exit early
+  if (xhr.length === 0 && !bool) {
+    expect(xhr.length).to.equal(0, `${url} wasn't requested`)
+    return
+  }
+  if (xhr.length > 0) {
+    const data = YAML.parse(yaml) as {
+      method: string,
+      searchParams: Record<string, string>,
+      body: Record<string, unknown>
+    }
+    const found = xhr.find((item: HistoryEntry) => {
+      return Object.entries(data).every(
+        ([key, value]) => {
+          switch (key) {
+            case 'method':
+              return item.request[key] === String(value)
+            case 'body':
+              return Object.entries(data[key]).every(([prop, value]) => {
+                return item.request[key][prop] === String(value)
+              })
+            case 'searchParams':
+              return Object.entries(data[key]).every(([key, value]) => {
+                const actual = item.url.searchParams.getAll(key)
+                // convert input to an array
+                const expected = Array.isArray(value) ? value : [value]
+                return expected.every((item) => {
+                  return actual.includes(String(item)) === true
+                })
               })
-            })
-            break
-        }
-      },
-    )
-  })
-})
-Then(/^the URL "(.*)" was requested with only$/, (url: string, exact: string, yaml: string) => {
-  cy.wait(`@${urls.get(url)}`).then((xhr) => {
-    const data = YAML.parse(yaml) as {method: string, searchParams: Record<string, string>, body: Record<string, unknown>}
-    Object.entries(data).forEach(
-      ([key, value]) => {
-        switch (key) {
-          case 'method':
-            expect(xhr.request[key]).to.equal(value)
-            break
-          case 'body': {
-            const bodyEntries = Object.entries(data[key])
-
-            bodyEntries.forEach(([prop, value]) => {
-              expect(xhr.request[key][prop]).to.equal(value)
-            })
-
-            // Asserts that the expected body data and the requested body data have the same amount of keys. If the previous assertion passed, that implies that the request body doesn’t have extraneous properties. This can be useful when utilizing PATCH requests.
-            if (exact) {
-              expect(Object.keys(xhr.request[key]).length).to.equal(bodyEntries.length)
-            }
-
-            break
-          }
-          case 'searchParams': {
-            const searchParamsEntries = Object.entries(data[key])
-            searchParamsEntries.forEach(([key, value]) => {
-              expect(xhr.request.query[key]).to.equal(value)
-            })
-
-            if (exact) {
-              expect(Object.keys(xhr.request.query[key]).length).to.equal(searchParamsEntries.length)
-            }
-
-            break
           }
-        }
-      },
-    )
-  })
```

Comment:
See https://github.com/kumahq/kuma-gui/issues/1142. The places where did use this where broken due to this step not working, and therefore commented out. These currently commented out tests will be replaced with the `was not request with` variant that was added recently.